HorizonNing/src/main/java/com/hubspot/horizon/ning/internal/CustomNingNameResolver.java

```diff
@@ -0,0 +1,46 @@
+package com.hubspot.horizon.ning.internal;
+
+import com.hubspot.horizon.CustomDnsResolver;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.List;
+import org.asynchttpclient.shaded.io.netty.resolver.InetNameResolver;
+import org.asynchttpclient.shaded.io.netty.util.concurrent.EventExecutor;
+import org.asynchttpclient.shaded.io.netty.util.concurrent.Promise;
+
+public class CustomNingNameResolver extends InetNameResolver {
+
+  private final CustomDnsResolver customDnsResolver;
+
+  public CustomNingNameResolver(
+    EventExecutor executor,
+    CustomDnsResolver customDnsResolver
+  ) {
+    super(executor);
+    this.customDnsResolver = customDnsResolver;
+  }
+
+  @Override
+  protected void doResolve(String inetHost, Promise<InetAddress> promise) {
+    try {
+      InetAddress[] addresses = customDnsResolver.resolve(inetHost);
```

Comment:
I don't think we want to block like this in our async client, can we make `CustomDnsResolver#resolve` return a `CompletableFuture<InetAddress[]>`? And then we can resolve the promise when the future resolves

Reply:
done!